src/aznamer.ts

```diff
@@ -24,52 +25,49 @@ export class AzNamer {
-            // return subOperationGroupName === "" ? "show" : subOperationGroupName + " " + "show";
-            // for show scenarios like kusto, if there's list, listbyresourcegroup, listsku, listskubyresource
-            // we should divide it into two groups
-            // group list contains list and listbyresourcegroup
-            // group listsku contains listsku and listskubyresource
-            // a temporary way is to treat the part after 'by' as parameter distinguish part and the part before by as command.
-            // the split is valid only the By is not first word and the letter before By is capital and the letter after By is lowercase \
-            const regex = /^(?<show>Get[a-zA-Z0-9]*)(?<by>By[A-Z].*)$/;
-            const groups = operationNameOri.match(regex);
-            let cmd = 'show';
-            if (groups && groups.length > 2) {
-                cmd = changeCamelToDash(groups[1]);
-            } else {
-                cmd = changeCamelToDash(operationNameOri);
-            }
-            cmd = cmd.replace(/^get/i, 'show');
-            return subOperationGroupName === '' ? cmd : subOperationGroupName + ' ' + cmd;
-        } else if (operationName.startsWith('list') && httpProtocol === 'get') {
+        function commandNameMap(type: string): string {
             // for list scenarios like kusto, if there's list, listbyresourcegroup, listsku, listskubyresource
             // we should divide it into two groups
             // group list contains list and listbyresourcegroup
             // group listsku contains listsku and listskubyresource
             // a temporary way is to treat the part after 'by' as parameter distinguish part and the part before by as command.
             // the split is valid only the By is not first word and the letter before By is capital and the letter after By is lowercase
-            const regex = /^(?<list>List[a-zA-Z0-9]*)(?<by>By[A-Z].*)$/;
+            // const regex = /^(?<$(type)>$(type)[a-zA-Z0-9]*)(?<by>By[A-Z].*)$/;
+            const regexStr =
+                '^(?<' + type + '>' + Capitalize(type) + '[a-zA-Z0-9]*)(?<by>By[A-Z].*)$';
+            const regex = new RegExp(regexStr);
             const groups = operationNameOri.match(regex);
-            let list = 'list';
+            let mtype = type;
             if (groups && groups.length > 2) {
-                list = changeCamelToDash(groups[1]);
+                mtype = changeCamelToDash(groups[1]);
             } else {
-                list = changeCamelToDash(operationNameOri);
+                mtype = changeCamelToDash(operationNameOri);
             }
-            return subOperationGroupName === '' ? list : subOperationGroupName + ' ' + list;
+            return subOperationGroupName === '' ? mtype : subOperationGroupName + ' ' + mtype;
+        }
+        let commandName = '';
+        if (operationName.startsWith('create') && httpProtocol === 'put') {
```

Comment:
Just curious, do you support httpmethod of 'post' or 'head' in CLI?

Reply:
yes. but create command has specific requirement in the httpProtocol. If some operation name create but with a post httpProtocol, it will use the operation original name as command name